uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-widget/leave-widget.component.js

```diff
@@ -82,27 +87,45 @@ define([
     }
 
     /**
-     * Watches for child components loading and ready events.
+     * Initializes watchers for child components and events that make the leave
+     * widget refresh.
      */
     function initWatchers () {
       $scope.$on('LeaveWidget::childIsLoading', childComponentIsLoading);
       $scope.$on('LeaveWidget::childIsReady', childComponentIsReady);
+      initWatchersForWidgetRefresh();
     }
 
     /**
-     * Loads absence types, the current absence period, and leave request
+     * Watches for events that make the leave widget refresh.
+     */
+    function initWatchersForWidgetRefresh () {
+      var listOfEvents = ['LeaveRequest::new', 'LeaveRequest::edit',
+        'LeaveRequest::deleted', 'LeaveRequest::updatedByManager',
+        'contract:created', 'contract:deleted', 'contract-refresh'];
+
+      listOfEvents.forEach(function (eventName) {
+        pubSub.subscribe(eventName, loadDependencies);
+      });
+    }
+
+    /**
+     * Loads the contact's job contract and if the contact has a job contract
+     * it then loads absence types, the current absence period, and leave request
      * statuses. When all dependencies are ready it sets loading component to
      * false.
      *
      * @return {Promise} - Returns an empty promise when all dependencies have
      * been loaded.
      */
     function loadDependencies () {
-      return $q.all([
-        loadAbsenceTypes(),
-        loadCurrentAbsencePeriod(),
-        loadLeaveRequestTypes()
-      ])
+      loadCurrentJobContract().then(function () {
```

Comment:
Does loadCurrentJobContract need to be loaded *before* a.types etc? Can it be loaded in parallel with others?

Reply:
As explained in the description, it is loaded before anything else because other requests are not needed if the contact has no contract.